File: tyreworks/__work/files/B2C DOTNET Version 4.0.2 - 2014-05-01/Examples/SendOrder.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Data.SqlClient;
using System.Data;

namespace TestingProject
{
    public class SendOrder
    {
        public int SendMidasOrder(int pOrderID)
        {
            try
            {
                MidasService.MidasOrderService service = new TestingProject.MidasService.MidasOrderService();

                MidasService.OrderDetails orderDetails = GetOrderDetailsFromDatabase(pOrderID);
                
                return service.CreateMidasOrder(pOrderID, "MyUserName", "MyPwd", orderDetails);
            }
            catch (Exception ex)
            {
                throw new Exception(string.Format("An error occurred sending the Order to Midas: {0}", ex.Message));
            }

        }

        public MidasService.OrderDetails GetOrderDetailsFromDatabase(int pOrderID)
        {
            try
            {
                MidasService.OrderDetails orderDetails = new MidasService.OrderDetails();
                
                using (SqlConnection sqlCon = new SqlConnection("MyConnectionString"))
                {
                    using (SqlCommand sqlCom = new SqlCommand("up_SelectOrderHeader", sqlCon))
                    {
                        sqlCom.CommandType = CommandType.StoredProcedure;
                        sqlCom.Parameters.Add("@OrderID", SqlDbType.Int).Value = pOrderID;
                        sqlCom.Connection.Open();

                        SqlDataReader sqlDr = sqlCom.ExecuteReader();

                        try
                        {
                            if (sqlDr.HasRows)
                            {
                                sqlDr.Read();

                                //Your MIDAS CusID and SiteID
                                orderDetails.CusID = Convert.ToInt32(sqlDr["DealerID"]);
                                orderDetails.SiteID = Convert.ToInt32(sqlDr["SiteID"]);

                                //Customer Details
                                orderDetails.Title = sqlDr["Title"].ToString();
                                orderDetails.CusFName = sqlDr["CusFName"].ToString();
                                orderDetails.CusLName = sqlDr["CusLName"].ToString();

                                //Customer Billing Address
                                orderDetails.BillAddr1 = sqlDr["BillAddr1"].ToString();
                                orderDetails.BillAddr2 = sqlDr["BillAddr2"].ToString();
                                orderDetails.BillAddr3 = sqlDr["BillAddr3"].ToString();
                                orderDetails.BillAddr4 = sqlDr["BillAddr4"].ToString();
                                orderDetails.BillPCode = sqlDr["BillPCode"].ToString();
                                orderDetails.BillPhone = sqlDr["BillPhone"].ToString();
                                orderDetails.BillMobile = sqlDr["BillMobile"].ToString();
                                orderDetails.BillEmail = sqlDr["UserID"].ToString();

                                //Delivery Customer Details 
                                orderDetails.DelvTitle = sqlDr["DelvTitle"].ToString();
                                orderDetails.DelvFName = sqlDr["DelvFName"].ToString();
                                orderDetails.DelvLName = sqlDr["DelvLName"].ToString();

                                //Delivery Address
                                orderDetails.DelvAddr1 = sqlDr["DelvAddr1"].ToString();
                                orderDetails.DelvAddr2 = sqlDr["DelvAddr2"].ToString();
                                orderDetails.DelvAddr3 = sqlDr["DelvAddr3"].ToString();
                                orderDetails.DelvAddr4 = sqlDr["DelvAddr4"].ToString();
                                orderDetails.DelvPCode = sqlDr["DelvPCode"].ToString();
                                orderDetails.DelvPhone = sqlDr["DelvPhone"].ToString();

                                //Dates and times of fitting (if applicable)
                                if (sqlDr["FittingDate"] != DBNull.Value)
                                {
                                    orderDetails.RegNum = sqlDr["RegNum"].ToString();
                                    orderDetails.FittingDate = Convert.ToDateTime(sqlDr["FittingDate"]);

                                    if (Convert.ToBoolean(sqlDr["IsAm"]))
                                    {
                                        orderDetails.StartTime = "08:30";
                                        orderDetails.EndTime = "11:59";
                                    }
                                    else
                                    {
                                        orderDetails.StartTime = "12:00";
                                        orderDetails.EndTime = "17:30";
                                    }

                                    orderDetails.Comments = sqlDr["Comments"].ToString();
                                }

                                //Details from Bank
                                orderDetails.AuthCode = sqlDr["AuthCode"].ToString();
                                orderDetails.CCTransID = sqlDr["PasRef"].ToString();

                                //Can be setup to be whatever you need it to be
                                orderDetails.InvoiceNo = Convert.ToInt32(sqlDr["ServerID"]).ToString("00") + sqlDr["OrderID"].ToString();
                                
                                //Extra details that may or may not be filled out
                                orderDetails.ContactPhoneHome = "";
                                orderDetails.ContactPhoneMobile = "";
                                orderDetails.ContactPhoneWork = "";
                                orderDetails.ContactEmail = "";


                            }
                            
                        }
                        finally
                        {
                            sqlDr.Close();
                        }

                        sqlCom.CommandText = "up_SelectOrderLines";
                        sqlDr = sqlCom.ExecuteReader(CommandBehavior.CloseConnection);

                        try
                        {
                            List<MidasService.OrderLine> orderLineList = new List<MidasService.OrderLine>();
                            

                            while (sqlDr.Read())
                            {
                                MidasService.OrderLine orderLine = new MidasService.OrderLine();

                                orderLine.InvoiceNo = orderDetails.InvoiceNo;
                                orderLine.Code = sqlDr["StockCode"].ToString();
                                orderLine.Qty = Convert.ToInt32(sqlDr["Quantity"]);
                                orderLine.Price = Convert.ToDouble(sqlDr["CostWithVAT"]); //Price per item with VAT

                                orderLineList.Add(orderLine);
                            }

                            orderDetails.Lines = orderLineList.ToArray();
                        }
                        finally
                        {
                            sqlDr.Close();
                        }

                    }

                    
                }

                return orderDetails;
            }
            catch (Exception ex)
            {
                throw new Exception(string.Format("An error occurred getting the Order Details: {0}", ex.Message));
            }
        }

    }
}

File: tyreworks/__serverside/java/Examples/SendOrder.php
<?php
require "./midasorder.php";
//This is a required file for the MIDaS ordering service.
//Example How to send an order into MIDaS Using PHP
//Create an OrderHeader array with the values of the order header
$OrderHeader = array
(
//Your OrderID
"OrderID" => "12234",
//Your MIDaS Siteid
"SiteID" => "40",
//Your MIDas CusID
"CusID" => "879",
//MIDaS UserName and Password
"UserName" => "Test",
"Password" => "simon",
//Customer Details
"Title" => "Mr",
"CusFName" => "Simon",
"CusLName" => "Woodland", 
//Customer Billing Address
"BillAddr1" => "Draycott Business Park",
"BillAddr2" => "Cam",
"BillAddr3" => "Dursley",
"BillAddr4" => "Glos",
"BillPCode" => "GL11 5DQ",
"BillPhone" => "01453 891000",
"BillMobile" => "",
"BillEmail" => "",
//Delivery Customer Details 
"DelvTitle" => "",
"DelvFName" => "", 
"DelvLName" => "",
//Delivery Address
"DelvAddr1" => "",
"DelvAddr2" => "",
"DelvAddr3" => "",
"DelvAddr4" => "",
"DelvPCode" => "",
"DelvPhone" => "",
//Customer Registration Number
"RegNum" => "ABC 123X",
//Fitting Date (yyyy-mm-dd-Thh:mm:ss)
"FittingDate"=> "2005-06-04T00:00:00",
//Customer Special Instructions/Comments
"Comments" => "", 
	//Credit Card Authorisation number
	"AuthCode" => "1234", 
	//Your Invoice Number
	"InvoiceNo" => "LINUX", 
	//Fitting time slot (e.g. can be used to donate if it a morning or afternoon slot) hh:mm
	"StartTime" => "09:00", //Fitting Start time
	"EndTime" => "13:00",  // Fitting End time
	//CreditCard Transaction ID
	"CCTransID" => "123-445-sds",
	//Contact Fields
	"ContactPhoneHome" => "12345678",
	"ContactPhoneMobile" => "12345677",
	"ContactPhoneWork" => "456789",
	"ContactEmail" => "somone@someone.com"			        		
	);
//Create a new MidasOrder Object
$MidasOrder = new MidasOrder;
//Put the OrderHeader into the MidasOrder
$MidasOrder->CreateOrderHeader($OrderHeader);
//Create An Array for the order lines
$OrderLine = array
 (
	//Your StockCode
	"Code" => "2155516HAVOCR85",
	"Qty"  => 2,
	"Price" => 48.81,
);
//Add this Line to the MidasOrder
$MidasOrder->AddOrderLine($OrderLine);
//Add More Lines to the MidasOrder
$OrderLine = array
 (	
	"Code" => "2155516VBRIER30",
	"Qty"  => 2,
	"Price" => 64.64,
);
//Add 2nd Line to the MidasOrder
$MidasOrder->AddOrderLine($OrderLine);
//Finally the Send the Order to MIDAS
try
{		
		$result = $MidasOrder->SendOrder();							if($result > 0)
{
		//Update your tables with MidasOrderID here.
		//Thank the customer for the order.
		echo("MIDaS OrderRef ".$result);
	}
else
{
		echo("MIDaS OrderRef ".$result);
		//Log, Send and Email etc here, and thank the customer for the order
	}
}
catch (Exception $e)
{
//Log, Send an Email etc here, and thank the customer for the order.
echo($e->getMessage());
}
?> 
